ClntIfaceMgr/ClntIfaceMgr.cpp

```diff
@@ -593,7 +593,7 @@ bool TClntIfaceMgr::modifyPrefix(int iface, SPtr<TIPv6Addr> prefix, int prefixLe
             subprefixLen = prefixLen;
         } else if (ifaceLst.size()<256) {
             subprefixLen = prefixLen + bit_shift;
-            int offset = prefixLen/bit_shift;
+            int offset = (prefixLen+bit_shift) / 8;
```

Comment:
Code  Pull requests 2  Pulse
Conversation	Commits	Changes
1 changed file
1 addition and 1 deletion
ClntIfaceMgr/ClntIfaceMgr.cpp
@@ -593,7 +593,7 @@ bool TClntIfaceMgr::modifyPrefix(int iface, SPtr<TIPv6Addr> prefix, int prefixLe
             subprefixLen = prefixLen;
         } else if (ifaceLst.size()<256) {
             subprefixLen = prefixLen + bit_shift;
-            int offset = prefixLen/bit_shift;
@Platinumghost
Platinumghost 
added a note 1 minute ago
@Platinumghost
Platinumghost 
added a note less than a minute ago
+all
 
Leave a comment
 Cancel  Add reply Start a review
+            int offset = (prefixLen+bit_shift) / 8;
 
Leave a comment
 Cancel  Start a review
             if (prefixLen%8 == 0) {
                 // that's easy, just put ID in the next octet
                 buf[offset] = (*i)->getID();
 
Leave a comment
 Cancel  Start a review
Submit your review
 
Leave a comment
Comment
Submit general feedback without explicit approval.
Approve
Submit feedback approving these changes.
Request changes
Submit feedback suggesting changes.
Submit review
 Desktop version